---
title: "SummarizedExperiment"
output: html_notebook
---


# Assays
The airway package contains an example dataset from an RNA-Seq experiment of read counts per gene for airway smooth muscles. These data are stored in a `RangedSummarizedExperiment` object which contains 8 different experimental and assays 64,102 gene transcripts.


```{r}
library(SummarizedExperiment)
data(airway, package="airway")
se <- airway
se
```

To retrieve the experiment data from a `SummarizedExperiment` object one can use the `assays()` accessor. An object can have multiple assay datasets each of which can be accessed using the `$` operator. The airway dataset contains only one assay (`counts`). Here each row represents a gene transcript and each column one of the samples.

```{r}
assays(se)$counts
```



## ‘Row’ (regions-of-interest) data

The `rowRanges()` accessor is used to view the range information for a `RangedSummarizedExperiment.` (Note if this were the parent `SummarizedExperiment` class we’d use `rowData()`). The data are stored in a `GRangesList` object, where each list element corresponds to **one gene** transcript and the ranges in each `GRanges` correspond to the **exons** in the transcript.


```{r}
rowRanges(se)
```

## ‘Column’ (sample) data

Sample meta-data describing the samples can be accessed using `colData()`, and is a `DataFrame` that can store any number of descriptive columns for each sample row.

```{r}
colData(se)
```

```{r}
metadata(se)
```


# Constructing a SummarizedExperiment

Often, `SummarizedExperiment` or `RangedSummarizedExperiment` objects are returned by functions written by other packages. However it is possible to create them by hand with a call to the `SummarizedExperiment()` constructor.

Constructing a `RangedSummarizedExperiment` with a `GRanges` as the *rowRanges* argument:

```{r}
nrows <- 200
ncols <- 6

# The data Matrix
counts <- matrix(runif(nrows * ncols, 1, 1e4), nrows)

# Metadata of the rows (in this case are genes and regions)
rowRanges <- GRanges(rep(c("chr1", "chr2"), c(50, 150)),
                     IRanges(floor(runif(200, 1e5, 1e6)), width=100),
                     strand=sample(c("+", "-"), 200, TRUE),
                     feature_id=sprintf("ID%03d", 1:200))

# Metadata of the columns (in this case are samples)
colData <- DataFrame(Treatment=rep(c("ChIP", "Input"), 3),
                     row.names=LETTERS[1:6])

# Construct the summarized Experiment
SummarizedExperiment(assays=list(counts=counts),
                     rowRanges=rowRanges, colData=colData)
```

A `SummarizedExperiment` can be constructed with or without supplying a `DataFrame` for the *rowData* argument:

```{r}
SummarizedExperiment(assays=list(counts=counts), colData=colData)
```



# TEST

```{r}
data('sacurine',package = 'ropls')

data <- sacurine$dataMatrix
sampleMetadata <- sacurine$sampleMetadata
variableMetadata <- sacurine$variableMetadata

# DATA
# Rownames == SAMPLES // Colnames == METABOLITES
data

# SAMPLE METADATA
# Rownames == SAMPLES // Colnames == METADATA
sampleMetadata

# VARIABLE METADATA
# Rownames == METABOLITES // Colnames == METADATA
variableMetadata

```

## Constructing a SummarizedExperiment

```{r}
# Make the rows of data as data cols
variableMetadata$annotation=rownames(variableMetadata)
rownames(variableMetadata)=1:nrow(variableMetadata)
colnames(data)=1:ncol(data)


# Create a DatasetExperiment object
DE <- DatasetExperiment(data = data,
                  sample_meta = sampleMetadata,
                  variable_meta = variableMetadata,
                  name = 'Sacurine dataset')

DE


```







